samples/archivist_story_runner.py

```diff
@@ -0,0 +1,305 @@
+"""
+
+import time
+from enum import Enum
+import argparse
+import yaml
+
+# remove warning from console
+import urllib3
+
+from archivist.compliance_polices import PolicyType
+from archivist.archivist import Archivist
+
+# stop printing to console that the request is insecure
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class Operation(Enum):
+    """
+    Operation are the allowed archivist operations to be run.
+    """
+
+    CREATE_ASSET = 1
+    CREATE_EVENT = 2
+    CREATE_COMPLIANCE_POLICY = 3
+    CHECK_COMPLIANCE = 4
+    DELETE_COMPLIANCE = 5
+
+
+def parse_arg(arg, key, default=None):
+    """
+    parse_arg parses a dictionary arg, and returns the value of a key.
+              if the key doesn't exist within the dictionary the default is returned.
+              if the default is None, then KeyError is raised.
+    """
+    try:
+        return arg[key]
+    except KeyError as ke:
+        if default is None:
+            raise ke
+
+        return default
+
+
+class ArchivistStoryRunner():
+    """
+    ArchivistStoryRunner takes a url, token_file and a yaml config_file.
+
+    The yaml config_file contains a list of `operations` to be performed serially, e.g.
+
+    ```
+    operations:
+
+        - operation: CREATE_ASSET
+            to_print: Create an empty radiation bag with id 1.
+            wait_time: 10
+            asset_id: radiation bag 1
+            behaviours:
+              - RecordEvidence
+            attributes:
+              radioactive: "true"
+              radiation_level: "0"
+              weight: "0"
+    ```
+
+    where:
+     - `operation` is the operation to perform (see Operation Enum for options).
+     - `to_print` is what to print to the console.
+     - `wait_time` is time to wait before running the operation.
+     - `asset_id` is the local reference to the asset, so can be referenced in other operations.
+     - `behaviours` are a list of asset behaviours
+     - `attributes are the asset's attributes
+
+    please see example yaml files for other operation examples.
+
+    To perform all the operations call the `run` method.
+    """
+
+    assets = dict()  # dict of assets created
+    events = list()  # list of events created
+    policies = dict()  # dict of policies created
+
+    def __init__(self, url, token_file, config_file):
+        """
+        constructor
+        :param url: the url of the archivist instance.
+        :param token_file: the token auth for the archivist instance.
+        :param config_file: the yaml config file listing all the operations to perform.
+        """
+
+        with open(token_file, mode="r") as tokenfile:
+            token = tokenfile.read().strip()
+
+            self.client = Archivist(url, auth=token, verify=False)
+
+        # read and save the config file
+        yaml_stream = open(config_file, 'r')
+        self.yaml_object = yaml.load(yaml_stream, Loader=yaml.SafeLoader)
+
+    def run(self):
```

Comment:
rename this to 

             __call__(self) 

and make the instance of the class directly callable.#

https://medium.com/swlh/callables-in-python-how-to-make-custom-instance-objects-callable-too-516d6eaf0c8d

Reply:
have made the class directly callable :)